source/docs/software/basic-programming/reading-stacktraces.rst

```diff
@@ -0,0 +1,553 @@
+
+            }
+
+   .. group-tab:: C++
+
+      .. code-block:: C++
+         :lineno-start: 17
+
+         class Robot : public frc::TimedRobot {
+            public:
+               void RobotInit() override {
+                  motorRef = &m_armMotor;
+                  motorRef->SetInverted(false);
+               }
+
+            private:
+               frc::PWMVictorSPX m_armMotor{0};
+               frc::PWMVictorSPX * motorRef;
+         };
+
+
+
+Divide by Zero
+^^^^^^^^^^^^^^
+
+It is not generally possible to divide an integer by zero, and expect reasonable results. Most processors (including the roboRIO) will raise an Unhandled Exception.
+
+For example, consider the following code:
+
+.. tabs::
+
+   .. group-tab:: Java
+
+      .. code-block:: Java
+          :lineno-start: 18
+
+            int armLengthRatio;
+            int elbowToWrist_in = 39;
+            int shoulderToElbow_in;
+
+            @Override
+            public void robotInit() {
+
+               armLengthRatio = elbowToWrist_in / shoulderToElbow_in;
+
+            }
+
+   .. group-tab:: C++
+
+      .. code-block:: C++
+          :lineno-start: 17
+
+            class Robot : public frc::TimedRobot {
+               public:
+               void RobotInit() override {
+                  armLengthRatio = elbowToWrist_in / shoulderToElbow_in;
+               }
+
+               private:
+                  int armLengthRatio;
+                  int elbowToWrist_in = 39;
+                  int shoulderToElbow_in;
+
+            };
+
+When run, you'll see output that looks like this:
+
+.. tabs::
+
+   .. group-tab:: Java
+
+      .. code-block:: text
+
+         ********** Robot program starting **********
+         Error at frc.robot.Robot.robotInit(Robot.java:25): Unhandled exception: java.lang.ArithmeticException: / by zero
+               at frc.robot.Robot.robotInit(Robot.java:25)
+               at edu.wpi.first.wpilibj.TimedRobot.startCompetition(TimedRobot.java:94)
+               at edu.wpi.first.wpilibj.RobotBase.runRobot(RobotBase.java:335)
+               at edu.wpi.first.wpilibj.RobotBase.lambda$startRobot$0(RobotBase.java:387)
+               at java.base/java.lang.Thread.run(Thread.java:834)
+
+         Warning at edu.wpi.first.wpilibj.RobotBase.runRobot(RobotBase.java:350): unexpected error has occurred, but yours did!
+         Error at edu.wpi.first.wpilibj.RobotBase.runRobot(RobotBase.java:352): The startCompetition() method (or methods called by it) should have handled the exception above.
+
+      Looking at the stack trace, we can see a ``java.lang.ArithmeticException: / by zero`` exception has occurred on line 25. If you look at the two variables which are used on the right-hand side of the ``=`` operator, you might notice one of them has not been initialized. This means its value is, by default, zero. And, the zero-value variable is used in the denominator of a division operation. Hence, the divide by zero error happens.
```

Comment:
I see here that you mention uninitialized ints will default to zero, so maybe you can disregard the comment above in Java. C++ comment still applies however.

Reply:
I went ahead and changed the error in coding to be something like:

```
variable = 0; //TODO

```

to imply the situation where someone put it there as placeholder but forgot to change it. That should hopefully sidestep the init question (as I believe the answer changes for java/c++, as well as windows/linux and stack vs. heap?)